src/game_state/Player.cpp

```diff
@@ -59,10 +65,19 @@ void Player::SetYVelocity(float y_velocity) {
 
 void Player::SetSpacebarDown(bool spacebar_down) {
   this->spacebar_down = spacebar_down;
+  if (!this->ground && spacebar_down && this->can_double_jump) {
+    this->is_double_jumping = true;
+    this->can_double_jump = false;
```

Comment:
Uhh idk how it would be double jumping immediately.. SetSpacebarDown only gets called when the spacebar is pressed, so it will only double jump once you release and press the space bar again.